ancpbids/lazy_load.py

```diff
@@ -0,0 +1,102 @@
+from enum import Enum, auto
+from typing import List, Union, Dict, Any
+from math import inf
+import sys
+import os
+
+class LazyLoadingMixin:
```

Comment:
This direction is great: use a Mixin class to implement lazy-loading.
Though, there is no need to have a format specific implementation. I think it should be sufficient to have only the mixin class. Need to think about how to make this a bit simpler.